frontend/ui/widget/booklist.lua

```diff
@@ -0,0 +1,94 @@
+local DocSettings = require("docsettings")
+local Menu = require("ui/widget/menu")
+
+local BookList = Menu:extend{
+    covers_fullscreen = true, -- hint for UIManager:_repaint()
+    is_borderless = true,
+    is_popout = false,
+    book_info_cache = {}, -- cache in the base class
+}
+
+function BookList:init()
+    self.title_bar_fm_style = not self.custom_title_bar
+    Menu.init(self)
+end
+
+-- BookInfo
+
+function BookList.setBookInfoCache(file, doc_settings)
```

Comment:
Everything (in the other changed modules) reads rather well.
This one is ok, but just mentionning some minor thought, in case you prefer these options:
`setBookInfoCache(doc_settings)` could read (a quick read) that we set the `doc_settings` as the cached thing.
`setBookInfoCacheFrom(doc_settings)` would make it clearer that we don't, but it's long and ugly.
`updateBookInfoCache(doc_settings)` would avoid the "set" feeling, and "update" is sufficiently opaque that we know we update the cache from the doc_settings.